docs/import.md

```diff
@@ -0,0 +1,137 @@
+## Automatic importation
+
+Automatic importation of files is one of the main feature of rest-on-couch. The idea
+is that by placing text files in specific folder the data can be imported automatically.
+
+As an example we can take a database called `eln`. On the file system we will have this
+kind of hierarchy:
+
+| path                                  | description                                                                          |
+| ------------------------------------- | ------------------------------------------------------------------------------------ |
+| rest-on-couch-home                    | rest-on-couch home of configurations file                                            |
+| rest-on-couch-home/config.js          | general configuration applying to all databases                                      |
+| rest-on-couch-home/eln/               | home of configuration of a specific database, here `eln`                             |
+| rest-on-couch-home/eln/config.js      | configuration of a specific database, here `eln`. Define here custom couchdb `views` |
+| rest-on-couch-home/eln/nmr/import.js  | script allowing to import automatically files and add metadata and attachments       |
+| rest-on-couch-home/eln/nmr/to_process | drop here the files to be processed                                                  |
+| rest-on-couch-home/eln/nmr/processed  | find here the processed files                                                        |
+| rest-on-couch-home/eln/nmr/errored    | find here the errored files                                                          |
+
+## import.js
+
+`import.js` will exports an async function that will be used to process files to import.
+`module.exports = async function nmrImport(importContext, result) {}`
+
+"Important note": filename HAS to be unique for a specific kind of data otherwise the existing file will be replaced !!
+This is true even if the field `reference` is different.
+
+The importation procedure has a special `importContext` in which the following variables are available:
```

Comment:
To be clearer:
"The import function takes two argument. The first is `importContext`, and allows to retrieve all information about the file being currently imported. It has the following properties and methods:"